backend/src/services/secret-v2-bridge/secret-v2-bridge-service.ts

```diff
@@ -1636,6 +1638,77 @@ export const secretV2BridgeServiceFactory = ({
     };
   };
 
+  const getSecretReferenceTree = async ({
+    environment,
+    secretPath,
+    projectId,
+    actor,
+    actorId,
+    actorOrgId,
+    secretName,
+    actorAuthMethod
+  }: TGetSecretReferencesTreeDTO) => {
+    const { permission } = await permissionService.getProjectPermission(
+      actor,
+      actorId,
+      projectId,
+      actorAuthMethod,
+      actorOrgId
+    );
+
+    ForbiddenError.from(permission).throwUnlessCan(
+      ProjectPermissionActions.Read,
+      subject(ProjectPermissionSub.Secrets, { environment, secretPath })
+    );
```

Comment:
Don't we need to check for permissions at each reference along the way in addition to the root secret?

Reply:
https://github.com/Infisical/infisical/pull/2603/files/39013cbeee4e1d013a4e13bc68b4903ce28bf299#diff-7d02b63b99e49c50295fdd39e013c8e9294316950497e86f31e56be915dfa8caR460